### PR TITLE
[Review] Request from 'schubi2' @ 'yast/yast-users/review_160830_checking_user_entries_in_autoyast'

### PR DESCRIPTION
**Pull request by schubi2**
Please review the following changes:
  * 524cc9a packaging
  * a1a7a69 checking defined users in AutoYaST profile

### CODE REVIEW

File: src/clients/users_auto.rb [hunk 1]
```diff
@@ -36,8 +36,24 @@
...
+
 module Yast
   class UsersAutoClient < Client
+
+    # Checking double user entries
+    # (double username or UID)
+    # @param [Array] users to check
+    def check_users(users)
+      if users.size > users.uniq { |u| u["username"]}.size
+        Report.Error(_("Found users in profile with equal <usersname>."))
```
Comment (imobachgs) [inline src/clients/users_auto.rb]: Typo? `<username>`

File: package/yast2-users.changes [hunk 2]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Tue Aug 30 10:37:18 CEST 2016 - schubi@suse.de
+
+- AutoYaST: Checking users entries for double username/UIDs.
```
Comment (imobachgs) [inline package/yast2-users.changes]: I would say `duplicated` instead of `double`.

File: src/clients/users_auto.rb (hunk 1 above)
Comment (schubi2) [inline src/clients/users_auto.rb]: Yes, as usual :-)

File: test/users_auto_test.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        let(:users) { YAML.load_file(FIXTURES_PATH.join("users_error.yml")) }
+
+        before do
+          allow(Yast::WFM).to receive(:Args).with(no_args).and_return([func,users])
+          allow(Yast::WFM).to receive(:Args).with(0).and_return(func)
+          allow(Yast::WFM).to receive(:Args).with(1).and_return(users)
+        end
+
+        it "report error" do
+          expect(Yast::Report).to receive(:Error).with("Found users in profile with equal <username>.")
```
Comment (imobachgs) [inline test/users_auto_test.rb]: This test may fail in systems with a different locale. You can use the `_` method here if you add these lines to `test_helper.rb`:

```ruby
RSpec.configure do |config|
  config.extend Yast::I18n  # available in context/describe
  config.include Yast::I18n # available in it/let/before/...
end
```

See https://github.com/yast/yast-packager/blob/master/test/test_helper.rb#L17 for an example.

File: src/clients/users_auto.rb [hunk 4]
```diff
@@ -36,8 +36,24 @@
...
+
+Yast.import "Report"
+
 module Yast
   class UsersAutoClient < Client
+
+    # Checking double user entries
+    # (double username or UID)
+    # @param [Array] users to check
+    def check_users(users)
```
Comment (imobachgs) [inline src/clients/users_auto.rb]: If that method is not part of the public API (and it's not), I would move it to the end and set it as `private`.

**Comment by imobachgs** [comment]
LGTM if Travis is happy.

**Pull request merged**